### PR TITLE
Adding TCU Debarred crawler

### PR DESCRIPTION
**Pull request by bgmello**
Very similar to https://github.com/opensanctions/opensanctions/pull/546. Just some minor changes on the URLs and entity types, which in this case can be either a company or a person.

This PR is related to the issue [opensanctions/crawler_planning#18](https://github.com/opensanctions/crawler-planning/issues/18). I changed the name from unfit to debarred because I think it makes more sense in this case.

Here is an unofficial link explaining the dataset: https://suporte.linkana.com/pt-BR/articles/3136996-lista-de-licitantes-inidoneos

### CODE REVIEW

**Comment by bgmello** [comment]
I branched off from [br-tcu-disqualified](https://github.com/opensanctions/opensanctions/pull/546) to use the new `clean_br_cnpj` and `clean_br_cpf` functions from helper. That's why it has the commits from that PR too

File: datasets/br/br_tcu_debarred/br_tcu_debarred.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+title: Brazil's TCU list of Debarred Bidders
+entry_point: crawler.py
+prefix: br-tcu-debarred
+coverage:
+  frequency: daily
+load_db_uri: ${OPENSANCTIONS_DATABASE_URI}
+summary: >
+  The TCU list of Debarred Bidders is a catalog of individuals and companies sanctioned by Brazil's Tribunal de Contas da União, prohibiting them from participating in public procurement processes.
+description: |
+  The TCU compiles a list of those declared debarred to participate in federal public
```
Comment (jbothma) [inline datasets/br/br_tcu_debarred/br_tcu_debarred.yaml:10]:
```suggestion
  The TCU compiles a list of those declared debarred from participating in federal public
```

File: datasets/br/br_tcu_debarred/br_tcu_debarred.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+title: Brazil's TCU list of Debarred Bidders
```
Comment (jbothma) [inline datasets/br/br_tcu_debarred/br_tcu_debarred.yaml:1]: just a reminder here too about the directory structure: br/tcu_debarred/br_tcu_debarred

File: datasets/br/br_tcu_debarred/br_tcu_debarred.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+load_db_uri: ${OPENSANCTIONS_DATABASE_URI}
+summary: >
+  The TCU list of Debarred Bidders is a catalog of individuals and companies sanctioned by Brazil's Tribunal de Contas da União, prohibiting them from participating in public procurement processes.
+description: |
+  The TCU compiles a list of those declared debarred to participate in federal public
+  administration tenders, under the terms of art. 46 of the Organic Law of the Court (Law 8.443/92).
+  Processes that are ongoing or whose decisions have not become final are not considered.
+publisher:
+  name: Tribunal de Contas da União (TCU)
+  url: https://portal.tcu.gov.br/inicio/
```
Comment (jbothma) [inline datasets/br/br_tcu_debarred/br_tcu_debarred.yaml:15]: Please add the english version of their name in publisher.description

File: datasets/br/br_tcu_debarred/br_tcu_debarred.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+title: Brazil's TCU list of Debarred Bidders
+entry_point: crawler.py
+prefix: br-tcu-debarred
```
Comment (jbothma) [inline datasets/br/br_tcu_debarred/br_tcu_debarred.yaml:3]:
```suggestion
prefix: br-tcu-debar
```

File: datasets/br/br_tcu_debarred/crawler.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    else:
+        context.log.warning("Entity type not defined by tax number")
+        tax_number = raw_tax_number
+        prefix = "br-unknown"
+
+    entity.id = context.make_slug(tax_number, prefix=prefix)
+    entity.add("name", input_dict["nome"])
+    entity.add("taxNumber", tax_number)
+    entity.add("country", "br")
+    entity.add("topics", "debarment")
```
Comment (jbothma) [inline datasets/br/br_tcu_debarred/crawler.py:44]: good. Please also add to the debarment collection

Comment (bgmello) [reply datasets/br/br_tcu_debarred/crawler.py]: Is `collection` another parameter for the entity different from the topics?

Comment (jbothma) [reply datasets/br/br_tcu_debarred/crawler.py]: oh, sorry, no a collection is a type of dataset that includes other datasets as children. 

This is a new section in the tutorial: https://zavod.opensanctions.org/tutorial/#add-your-crawler-to-a-collection

File: datasets/br/tcu_debarred/crawler.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    if len(raw_tax_number) == 18:
+        entity.add_schema("Company")
+        tax_number = CNPJ.normalize(raw_tax_number)
+        prefix = "br-cnpj"
+
+    # If it's length 14 with punctuation, then it is a CPF (XXX.XXX.XXX-XX)
+    elif len(raw_tax_number) == 14:
+        entity.add_schema("Person")
+        tax_number = CPF.normalize(raw_tax_number)
+        prefix = "br-cpf"
```
Comment (jbothma) [inline datasets/br/tcu_debarred/crawler.py:33]: you'll need to handle normalize returning None here - perhaps you can use normalize instead of the length condition?

Comment (jbothma) [reply datasets/br/tcu_debarred/crawler.py:33]: oh sorry I see how it's used now

File: zavod/zavod/helpers/__init__.py [hunk 7]
```diff
@@ -38,6 +38,7 @@ def crawl(context: Context) -> None:
 from zavod.helpers.positions import make_position, make_occupancy
 from zavod.helpers.text import clean_note, is_empty, remove_bracketed
 from zavod.helpers.text import multi_split
+from zavod.helpers.text import clean_br_cnpj, clean_br_cpf
```
Comment (jbothma) [inline zavod/zavod/helpers/__init__.py:41]: this should drop away, right?

**Comment by jbothma** [comment]
Thank you!
